### PR TITLE
Refactoring the tests for modeling.py

### PR DESCRIPTION
**Pull request by matthewkirby**
The tests as they stand are very messy, all in the base scope, and not well-documented or complete. In this PR `test_modeling.py` will be brought up to the standards of the rest of the code base. 

Out of curiosity, cover was 54.294% after commenting out `test_modeling.py`. I am curious what it is up to afterwards.

### CODE REVIEW

**Comment by matthewkirby** [comment]
#### In `cclify_astropy_cosmo()`
- Checks if the passed type is `astropy.cosmology.core.FlatLambdaCDM` else just return the thing. 
- This is bad. What if someone passes in something random? A cosmology object from some other code? It always just spits it back out and says "nice job".
- [ ] Why doesn't this support other astropy cosmology objects? It is easy to do, just check type against the base parent class.
- [ ] `elif type (is CCL object)` For now this is just dict
- [ ] Then we else raise exception.
- [ ] Also need to check to ensure that the astropy obj has the variable set. If it doesn't, it doesn't throw an error, just returns a None

**Comment by matthewkirby** [comment]
#### In `_get_a_from_z()` and vice-versa
- [ ] Currently not checking domain. Probably fine without it but its one line and could help debug down the line so I am going to add it in.

**Comment by matthewkirby** [comment]
#### In !_ct_omega_m_fix`
- This function evolves Omega_m in redshift. The intent is to pass this evolved quantity to cluster_toolkit to approximate the redshift evolution.
- I am going to rename this to `_evolve_omega_m` to more desciptive about what it does. 
- I am fairly sure this function is wrong. 
    - Currently: Omega_m(z) = Omega_m(z=0) * (1+z)^3
    - This should be: Omega_m(z) = rho_m(z)/rho_crit(z) = Omega_m(z=0) * (1+z)^3 / E(z)^2
        - This is just from rho_m(z) ~ (1+z)^3 and rho_crit(z) ~ H(z)^2

**Comment by matthewkirby** [comment]
After discussion with Michel, this function isn't computing Omega_m(z) as was claimed. The goal of this function is to trick cluster_toolkit into redshift evolution. Cluster_toolkit says rho_m = Omega_m * rho_c, all at z=0. So by passing `Omega_m * (1+z)**3`, we get the redshift evolution of rho_m. 

- [ ] Make new module `cluster_toolkit_patches.py` and relocate this function there. This is to avoid anyone actually using this function.

**Comment by matthewkirby** [comment]
This PR is ready for review, but please do not merge it. I want to get Michel's PRs merged first so that I can personally merge the validation tests into these tests.

File: tests/test_modeling.py [hunk 1]
```diff
@@ -1,83 +1,349 @@
...
+                                                    z_source=cfg['TEST_CASE']['z_source']),
+                    cfg['TEST_CASE']['nc_Sigmac'], 1.0e-8)
 
-def test_DeltaSigma():
-    tst.assert_equal(DeltaSigma[-1], DeltaSigma_one)
-    assert(np.all(DeltaSigma > 0.))
-    assert(DeltaSigma_one > 0.)
 
-Sigmac = clmm.get_critical_surface_density(cosmo_ccl, z_cluster=1.0, z_source=2.0)
+def helper_physics_functions(func):
```
Comment (m-aguena) [inline tests/test_modeling.py]: same as `helper_profiles`

File: clmm/cluster_toolkit_patches.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Parameters
+    ----------
+    omega_m : float
+        Mean matter density at z=0 in units of the critical density
+    redshift: array_like
+        Redshift
+
+    Returns
+    -------
+    omega_m : float
```
[the remark this answers is not in the document]
Comment (matthewkirby) [reply clmm/cluster_toolkit_patches.py]: This is the same way its done everywhere else in the code. Do you think we should have it the way you are suggesting?

[the remark this answers is not in the document]
Comment (matthewkirby) [reply clmm/cluster_toolkit_patches.py]: Reasoning behind what I wrote is that now a user knows exactly that this is returning omega_m.

File: tests/test_modeling.py [hunk 3]
```diff
@@ -1,83 +1,349 @@
...
+    # Check output including: float, list, ndarray
+    assert_allclose(md.get_reduced_shear_from_convergence(shear[0], convergence[0]),
+                    truth[0], **TOLERANCE)
+    assert_allclose(md.get_reduced_shear_from_convergence(shear, convergence),
+                    truth, **TOLERANCE)
+    assert_allclose(md.get_reduced_shear_from_convergence(np.array(shear), np.array(convergence)),
+                    np.array(truth), **TOLERANCE)
+
+
+def helper_profiles(func):
```
[the remark this answers is not in the document]
Comment (matthewkirby) [reply tests/test_modeling.py]: I added a little more description.

File: tests/test_modeling.py (hunk 1 above)
Comment (matthewkirby) [reply tests/test_modeling.py]: I added a little more description.

**Comment by matthewkirby** [comment]
@m-aguena I went through and addressed your three comments. I also tested running the docs (uncommenting the `--execute` line in `docs/conf.py` and running the `./update_docs` script). I found that this crashed for two reasons.

1. The API updates that I made in this PR were not propagated to the example notebooks.
2. The example notebooks had metadata added which referenced an unknown kernel. I now set the kernel specifically that I want nbconvert to use. While slightly outside the scope of this PR, I am just going to make the change and force merge it if that would be better.

File: clmm/cluster_toolkit_patches.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-aguena) [reply clmm/cluster_toolkit_patches.py]: My bad, I hadn't noticed we were using this formalism. I don't have a strong opinion on which to use as long as the documentation of the whole code is consistent.

**Comment by m-aguena** [comment]
It looks good to me. @aimalz @cavestruz If both of you agree, feel free to merge it. If no one else looks at it in a week, I will merge it myself.

**Comment by matthewkirby** [comment]
When you do make sure you do the steps in the other PR! Let’s make this
v0.1.2

**Pull request merged**